x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.ts

```diff
@@ -393,20 +468,54 @@ export async function getLogstashStats(
     plugins: {},
   };
 
-  await fetchLogstashStats(callCluster, clusterUuids, start, end, options);
-  await Promise.all(
-    clusterUuids.map(async (clusterUuid) => {
-      if (options.clusters[clusterUuid] !== undefined) {
-        await fetchLogstashState(
-          callCluster,
-          clusterUuid,
-          options.allEphemeralIds[clusterUuid],
-          start,
-          end,
-          options
-        );
-      }
-    })
-  );
+  // if index name contains '-mb', means metricbeat based monitoring
+  // filter_path and collapse fields in the queries differ on metricbeat vs. self-monitoring
+  // note: agent driven LS monitoring indices pattern differ ".ds-metrics-logstash*"
+  for (const clusterUuid of clusterUuids) {
+    const isSelfMonitoring: boolean = await isLogstashSelfMonitoring(callCluster, clusterUuid);
+
+    await fetchLogstashStats(callCluster, clusterUuids, start, end, options, isSelfMonitoring);
```

Comment:
We're calling this method in every iteration even when we're passing all `clusterUuids`.

I wonder if we could either call `fetchLogstashStats` for both values of `isSelfMonitoring` and discard the _nulls_ (we won't need `isLogstashSelfMonitoring` then because we know which clusterUuids are coming from each call).

Or we could group the clusters by `isLogstashSelfMonitoring` on each, and call `fetchLogstashStats` for each group.

WDYT?

Reply:
Ah, silly me. This should be a `clusterUuid` (`fetchLogstashStats(callCluster, clusterUuid...)`). Thanks for catching up, will update.